config/application.rb

```diff
@@ -114,13 +114,6 @@ class Application < Rails::Application
     # on the top to bottom order of initializer calls in the file.
     # Because this is easy to mess up, keep your initializers in order.
     # register plugins even before loading settings, as plugins can bring their own settings
-    initializer :register_vmdb_plugins, :before => :load_vmdb_settings do
-      Rails.application.railties.each do |railtie|
-        next unless railtie.class.name.start_with?("ManageIQ::Providers::") || railtie.try(:vmdb_plugin?)
-        Vmdb::Plugins.instance.register_vmdb_plugin(railtie)
```

Comment:
> reloaded in production 

Why would anything be reloaded in production? And even if, by moving the logic into the `Plugins#initialize` it is always initiliazed correctly vs having it being setup in an initializer, which only is executed at boot time.

> providing its own settings.yml, those have to be in place before the :load_vmdb_settings initializer is called

But they are. Because they are in `Plugins#initialize` - which ultimately makes sure, the plugins are registered __before__ something queries them...

So, I think my response answers the question 🤷‍♂️